canal/dump.go

```diff
@@ -113,6 +114,19 @@ func (h *dumpParseHandler) Data(db string, table string, values []string) error
 	return h.c.eventHandler.OnRow(events)
 }
 
+func (h *dumpParseHandler) Table(db string, query string) error {
+	if err := h.c.ctx.Err(); err != nil {
+		return err
+	}
+
+	event := &replication.QueryEvent{
+		Schema: []byte(db),
+		Query:  []byte(query),
+	}
+
+	return h.c.eventHandler.OnDDL(nil, mysql.Position{}, event)
```

Comment:
I don't really like that we don't pass `header` and `nextPos` to `OnDDL`. If a zero `nextPos` looks safe, then a nil `header` may break external code.

Reply:
There is no header can be init here, nil is reasonable, handler should check pointer before use like OnPosSynced in dump().

Reply:
Add nextPos in [7344273](https://github.com/go-mysql-org/go-mysql/pull/767/commits/73442733bb257815fcf57c57a96c556e1054ea7e), no header info provided here, it's OK to set nil pointer like OnPosSynced in dump(), user should check header before use in handler func.